### PR TITLE
Add functional tests for additional SeatNonBid codes

### PR DESCRIPTION
**Pull request by osulzhenko**
### 🔧 Type of changes
- [ ] new bid adapter
- [ ] update bid adapter
- [ ] new feature
- [ ] new analytics adapter
- [ ] new module
- [ ] bugfix
- [ ] documentation
- [ ] configuration
- [x] tech debt (test coverage, refactorings, etc.)

### ✨ What's the context?

What's the context for the changes? Are there any

### 🧠 Rationale behind the change

Why did you choose to make these changes? Were there any trade-offs you had to consider?

### 🧪 Test plan

How do you know the changes are safe to ship to production?

### 🏎 Quality check

- [ ] Are your changes following [our code style guidelines](https://github.com/prebid/prebid-server-java/blob/master/docs/developers/code-style.md)?
- [ ] Are there any breaking changes in your code?
- [ ] Does your test coverage exceed 90%?
- [ ] Are there any erroneous console logs, debuggers or leftover code in your changes?

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/tests/SeatNonBidSpec.groovy [hunk 1]
```diff
@@ -70,13 +80,69 @@ class SeatNonBidSpec extends BaseSpec {
...
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        and: "Account in the DB"
+        def accountConfig = new AccountConfig(auction: new AccountAuctionConfig(bidValidations:
+                new AccountBidValidationConfig(bannerMaxSizeEnforcement: ENFORCE)))
+        def account = new Account(status: ACTIVE, uuid: bidRequest.accountId, config: accountConfig)
+        accountDao.save(account)
+
+
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/SeatNonBidSpec.groovy:129]: Empty line

File: src/test/groovy/org/prebid/server/functional/tests/SeatNonBidSpec.groovy [hunk 2]
```diff
@@ -70,13 +80,69 @@ class SeatNonBidSpec extends BaseSpec {
...
+        def seatNonBid = response.ext.seatnonbid[0]
+        assert seatNonBid.seat == GENERIC.value
+        assert seatNonBid.nonBid[0].impId == bidRequest.imp[0].id
+        assert seatNonBid.nonBid[0].statusCode == RESPONSE_REJECTED_INVALID_CREATIVE_SIZE
+    }
+
+    def "PBS should populate seatNonBid when returnAllBidStatus=true and requested bidder responded with invalid creative size status code2"() {
+        given: "Default bid request with returnAllBidStatus"
+        def bidRequest = requestWithAllBidStatus.tap {
+            imp.ext.first
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/SeatNonBidSpec.groovy:145]: didn't get it

**Review by marki1an — CHANGES_REQUESTED** [review]
Add test for this one code 201 Request Blocked - Unsupported Channel (app/site/dooh) - if the request was not sent to the bidder because they don’t support dooh or app.

File: src/test/groovy/org/prebid/server/functional/tests/SeatNonBidSpec.groovy [hunk 3]
```diff
@@ -85,6 +151,66 @@ class SeatNonBidSpec extends BaseSpec {
...
+        assert seatNonBid.nonBid[0].statusCode == RESPONSE_REJECTED_INVALID_CREATIVE_SIZE
+    }
+
+    def "PBS should populate seatNonBid when returnAllBidStatus=true and requested bidder responded with not secure status code"() {
+        given: "PBS with secure-markup enforcement"
+        def pbsService = pbsServiceFactory.getService(["auction.validations.secure-markup": ENFORCE.value])
+
+        and: "A bid request with secure and returnAllBidStatus flags set"
+        def bidRequest = requestWithAllBidStatus.tap {
+            imp[0].secure = 1
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/SeatNonBidSpec.groovy:179]: And let's make it enum where 0 = non-secure, 1 = secure.

**Review by marki1an — CHANGES_REQUESTED** [review]

**Comment by osulzhenko** [comment]
> Add test for this one code 201 Request Blocked - Unsupported Channel (app/site/dooh) - if the request was not sent to the bidder because they don’t support dooh or app.

Unfortunately, our current Java implementation does not support blocking requests based on channels like app, site, or dooh.

**Review by marki1an — CHANGES_REQUESTED** [review]

File: src/test/groovy/org/prebid/server/functional/model/config/Ortb2BlockingAttributes.groovy [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package org.prebid.server.functional.model.config
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategies
+import com.fasterxml.jackson.databind.annotation.JsonNaming
+import groovy.transform.ToString
+
+@ToString(includeNames = true, ignoreNulls = true)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy)
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/model/config/Ortb2BlockingAttributes.groovy:8]: We can remove @JsonNaming

File: src/test/groovy/org/prebid/server/functional/model/config/Ortb2BlockingConfig.groovy [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package org.prebid.server.functional.model.config
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategies
+import com.fasterxml.jackson.databind.annotation.JsonNaming
+import groovy.transform.ToString
+
+@ToString(includeNames = true, ignoreNulls = true)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy)
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/model/config/Ortb2BlockingConfig.groovy:8]: Same here we can remove @JsonNaming

**Review by marki1an — CHANGES_REQUESTED** [review]

**Review by marki1an — APPROVED** [review]

**Pull request merged**